rest-service/manager_rest/deployment_update/manager.py

```diff
@@ -295,7 +299,15 @@ def commit_deployment_update(self,
         )
 
         # Update deployment attributes in the storage manager
-        deployment = self.sm.get(models.Deployment, dep_update.deployment_id)
+        parents_labels = []
+        if labels_to_create:
```

Comment:
This validation should be done before we check for preview https://github.com/cloudify-cosmo/cloudify-manager/blob/master/rest-service/manager_rest/deployment_update/manager.py#L254. 

Otherwise, the user might get a "happy" preview and then a failure in the parents labels verification.

Reply:
I think that validation is not needed since the validation happened on blueprint upload